scripts/RemoveBadSitesByID.wdl

```diff
@@ -0,0 +1,298 @@
+	>>>
+
+	runtime {
+docker: "us.gcr.io/broad-gatk/gatk:4.2.3.0"
+		disks: "local-disk " + disk_size + "  HDD"
+		memory: "16 GB"
+	}
+
+	output {
+		File par1_vcf = "PAR1.~{name}.vcf.gz"
+		File par1_vcf_index = "PAR1.~{name}.vcf.gz.tbi"
+		File par2_vcf = "PAR2.~{name}.vcf.gz"
+		File par2_vcf_index = "PAR2.~{name}.vcf.gz.tbi"
+		File non_par_vcf = "NON_PAR.~{name}.vcf.gz"
+		File non_par_vcf_index = "NON_PAR.~{name}.vcf.gz.tbi"
+	}
+}
+
+task BuildM3VCF {
+	input {
+		File vcf
+		File vcf_index
+	}
+
+	String name = basename(vcf, ".vcf.gz")
+
+	command <<<
+		/Minimac3Executable/bin/Minimac3 --refHaps ~{vcf} --processReference --prefix ~{name}.cleaned --rsid
+	>>>
+
+	runtime {
+		docker: "quay.io/ckachuli/minimac3@sha256:9b50682305869b103f2493179c9ea6ee618bd65d96bea17192dc609ce724c586"
+		memory: "16 GB"
+		disks: "local-disk 100 HDD"
+	}
+
+	output {
+		File out_m3vcf = "~{name}.cleaned.m3vcf.gz"
+	}
+}
+
+task VcfToBcf {
+	input {
+		File vcf
+	}
+
+	String name = basename(vcf, ".vcf.gz")
+
+	command <<<
+
+		bcftools view ~{vcf} -O b -o ~{name}.bcf
+		bcftools index ~{name}.bcf
+	>>>
+
+	runtime {
+		docker: "biocontainers/bcftools:v1.9-1-deb_cv1"
+		memory: "3 GiB"
+		disks: "local-disk 100 HDD"
+	}
+
+	output {
+		File out_bcf = "~{name}.bcf"
+		File out_bcf_index = "~{name}.bcf.csi"
+	}
+}
+
+task RemoveBadSitesFromVcf {
+	input {
+		File vcf
+		File bad_sites_list
+		String basename
+
+	}
+
+	parameter_meta {
+		vcf : {
+			localization_optional : true
+		}
+	}
+
+	command <<<
+		set -xeuo pipefail
+
+		cp ~{bad_sites_list} bad_sites.list
+
+		gatk SelectVariants -V ~{vcf} --exclude-ids bad_sites.list -O ~{basename}.cleaned.vcf.gz
+	>>>
+
+	runtime {
+                docker: "us.gcr.io/broad-gatk/gatk:4.2.3.0"
+		disks: "local-disk 100 HDD"
+		memory: "16 GB"
+	}
+
+	output {
+		File out_vcf = "~{basename}.cleaned.vcf.gz"
+		File out_vcf_index = "~{basename}.cleaned.vcf.gz.tbi"
+	}
+}
+
+task UpdateVariantIds {
+
+  input {
+    File vcf
+    String basename
+    Int disk_space =  3*ceil(size(vcf, "GB"))
+  }
+
+  command <<<
+    bcftools annotate --set-id '%CHROM\:%POS\:%REF\:%FIRST_ALT' ~{vcf} -O z -o ~{basename}.vcf.gz
+  >>>
+
+  output {
+    File output_vcf = "~{basename}.vcf.gz"
+  }
+
+  runtime {
+    docker: "skwalker/imputation:with_vcftools"
+    disks: "local-disk " + disk_space + " HDD"
+    memory: "16 GB"
+  }
+}
+
+task SortVariantIds {
+  input {
+    File vcf
+    String basename
+    Int disk_space =  3*ceil(size(vcf, "GB"))
+  }
+
+  command <<<
+    # what better way is there to do this I really don't know
+    zcat ~{vcf} | awk -v OFS='\t' '{split($3, n, ":"); if ( !($1 ~ /^"#"/) && n[4] < n[3])  $3=n[1]":"n[2]":"n[4]":"n[3]; print $0}' | bgzip -c > ~{basename}.vcf.gz
+    bcftools index -t ~{basename}.vcf.gz
+  >>>
+
+  output {
+    File output_vcf = "~{basename}.vcf.gz"
+    File output_vcf_index = "~{basename}.vcf.gz.tbi"
+
+  }
+
+  runtime {
+    docker: "skwalker/imputation:with_vcftools"
```

Comment:
is this docker widely available? If not it looks like any bcftoools docker might work